From eac58f720ad0b7727f966947ad9ab6dd3be058a4 Mon Sep 17 00:00:00 2001
From: Vladimir Skvortsov <vskvortsov@emcraft.com>
Date: Fri, 18 Sep 2026 12:54:43 +0200
Subject: [PATCH] imx6sx-blaze: RMII clocking for ENET1

board_init() calls setup_fec(), which sets ENET1 up for the SABRE-SD's RGMII
PHY: GPR1[13] cleared and a 125 MHz anatop clock. ENET1 here is RMII with the
50 MHz reference on the pad, so select that and run the anatop clock at 50 MHz.
ENET2 is left as it was.

Upstream-Status: Inappropriate [board-specific]
Signed-off-by: Vladimir Skvortsov <vskvortsov@emcraft.com>
---
 board/freescale/mx6sxsabresd/mx6sxsabresd.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/board/freescale/mx6sxsabresd/mx6sxsabresd.c b/board/freescale/mx6sxsabresd/mx6sxsabresd.c
index 137d7af..50091f3 100644
--- a/board/freescale/mx6sxsabresd/mx6sxsabresd.c
+++ b/board/freescale/mx6sxsabresd/mx6sxsabresd.c
@@ -167,13 +167,14 @@ static int setup_fec(void)
 	struct anatop_regs *anatop = (struct anatop_regs *)ANATOP_BASE_ADDR;
 	int reg, ret;
 
-	/* Use 125M anatop loopback REF_CLK1 for ENET1, clear gpr1[13], gpr1[17]*/
-	clrsetbits_le32(&iomuxc_regs->gpr[1], IOMUX_GPR1_FEC1_MASK, 0);
+	/* ENET1 is RMII: take the 50 MHz reference from the pad, gpr1[13] set. */
+	clrsetbits_le32(&iomuxc_regs->gpr[1], IOMUX_GPR1_FEC1_MASK,
+			IOMUX_GPR1_FEC1_CLOCK_MUX2_SEL_MASK);
 
 	/* Use 125M anatop loopback REF_CLK1 for ENET2, clear gpr1[14], gpr1[18]*/
 	clrsetbits_le32(&iomuxc_regs->gpr[1], IOMUX_GPR1_FEC2_MASK, 0);
 
-	ret = enable_fec_anatop_clock(0, ENET_125MHZ);
+	ret = enable_fec_anatop_clock(0, ENET_50MHZ);
 	if (ret)
 		return ret;
 
-- 
2.43.0

